RuntimeDetour/Platforms/Runtime/DetourRuntimeNETCorePlatform.cs

```diff
@@ -1,24 +1,158 @@
+        //      pop     eax           ; pop return adress
+        //      xchg    [esp+4], eax  ; exchange return adress with second arg (leaving return address under the parameters)
+        //      push    eax           ; push arg
+        //      lea     ebx, [esp+4]  ; store expected resulting stack pointer in nonvolatile register
+        //      call    <fptr>        ; call the function
+        //      cmp     esp, ebx      ; check if the stack pointer matches what we expected for a 2 arg call
+        //      je      .ret          ; if it matched, we're done so return
+        //      pop     ebx           ; otherwise it failed, so pop the other argument before returning
+        //    .ret:
+        //      ret     0
+        //
+        // This is to compensate for the fact that thiscall (and stdcall) on MSVC requires that the callee cleans up the stack according to its
+        //   arguments, so if a method takes only the this pointer as an argument, it will fail to pop the other argument and leave the stack
+        //   in a broken state.
+        //
+
+        private static void CallGetJitGuid(IntPtr jit, int index, out Guid guid) {
+            d_getVersionIdentifier getVersionIdentifier = ReadObjectVTable(jit, index)
+                .AsDelegate<d_getVersionIdentifier>();
+            getVersionIdentifier(jit, out guid);
+        }
+
+        // FIXME: .NET 5 has this method at index 2; how do we identify this?
+        private const int vtableIndex_ICorJitCompiler_getVersionIdentifier = 4;
+        private const int vtableIndex_ICorJitCompiler_getVersionIdentifier_net5 = 2;
+        [UnmanagedFunctionPointer(CallingConvention.ThisCall)]
+        private delegate void d_getVersionIdentifier(
+            IntPtr thisPtr, // ICorJitCompiler*
+            out Guid versionIdentifier
+            );
+
+        protected const int vtableIndex_ICorJitCompiler_compileMethod = 0;
+
+        protected static unsafe IntPtr* GetVTableEntry(IntPtr @object, int index)
+            => (*(IntPtr**) @object) + index;
+        protected static unsafe IntPtr ReadObjectVTable(IntPtr @object, int index)
+            => *GetVTableEntry(@object, index);
+
+        protected override void DisableInlining(MethodBase method, RuntimeMethodHandle handle) {
+            // https://github.com/dotnet/runtime/blob/89965be3ad2be404dc82bd9e688d5dd2a04bcb5f/src/coreclr/src/vm/method.hpp#L178
+            // mdcNotInline = 0x2000
+            // References to RuntimeMethodHandle (CORINFO_METHOD_HANDLE) pointing to MethodDesc
+            // can be traced as far back as https://ntcore.com/files/netint_injection.htm
+
+            // FIXME: Take a very educated guess regarding the offset to m_wFlags in MethodDesc.
+        }
+
+        public static readonly Guid Core31Jit = new Guid("d609bed1-7831-49fc-bd49-b6f054dd4d46");
+        public static readonly Guid Net50p4Jit = new Guid("6ae798bf-44bd-4e8a-b8fc-dbe1d1f4029e");

```

Comment:
In my opinion it'd make more sense to move those GUIDs to the respective implementation classes to prevent this class from blowing up any further 😅

Reply:
Done, but the resulting check code is a little bit long for my liking. I don't think there's a way to make it better though.